### PR TITLE
Fix manual competition creation & selenium test it

### LINKED ISSUE #422: [Bug] Can't create a competition using the online editor
I can use the online editor to create all the parts of a competition (the tabs "Participation", "Pages", "Phases", "Leaderboard" are with a green check)
screenshot-2020-06-16_-_22_33_10_-_Details
but I can't add a new administrator
screenshot-2020-06-16_-_22_33_10_-_Administrator
and I can't click on the "Save" button.
 From what I see on the online editor, all the mandatory fields are filled but I supposed that because the "Administrators" field is not with a green check, I can't save.

### PR DESCRIPTION
**Pull request by brendon-lind**
# @ mention of reviewers
@ckcollab 

# A brief description of the purpose of the changes contained in this PR.
Couldn't add a collaborator to a new manually created competition.
New issue discovered that the comp details page was not validating. Issued fixed 
Full selenium test for manual creation of a competition.
# Issues this PR resolves
#422 

# A checklist for hand testing
- [ ] Manually create a competition and go to administrator tab. Add a administrator.
- [ ] Check to make sure selenium tests passing.

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

File: src/static/riot/competitions/editor/_phases.tag [hunk 1]
```diff
@@ -62,10 +62,10 @@
             Edit phase
         </div>
         <div class="content">
-            <form class="ui form" ref="form">
+            <form selenium="phase-form" class="ui form" ref="form">
                 <div class="field required">
                     <label>Name</label>
-                    <input name="name">
+                    <input selenium="name1" name="name">
```
Comment (ckcollab) [inline src/static/riot/competitions/editor/_phases.tag:68]: This is extraneous? We should use as few attributes as possible, you should be able to look this up based on the above selenium attribute?

File: src/tests/functional/test_competitions.py [hunk 2]
```diff
@@ -44,3 +50,91 @@ def test_upload_v15_competition(self):
...
+        self.find('input[ref="logo"]').send_keys(os.path.join(self.test_files_dir, '3.png'))
+        self.find('input[ref="docker_image"]').send_keys('docker_image')
+
+        # Participation Tab
+        self.find('a[data-tab="participation"]').click()
+        self.execute_script('$("textarea[ref=\'terms\']")[0].EASY_MDE.value("pArTiCiPaTe")')
+        sleep(2)
+        self.find('input[selenium="auto-approve"]').click()
+
+        # Pages Tab
```
Comment (ckcollab) [inline src/tests/functional/test_competitions.py:91]: Nice comments and structure

File: src/tests/utils.py [hunk 3]
```diff
@@ -187,12 +187,13 @@ def remove_all_competition_data(self):
 
         # tasks and solutions
         for task in Task.objects.all():
-            created_files += [
-                task.ingestion_program.data_file.name,
-                task.scoring_program.data_file.name,
-                task.input_data.data_file.name,
-                task.reference_data.data_file.name,
-            ]
+            if competition.bundle_dataset is not None:
```
Comment (ckcollab) [inline src/tests/utils.py:190]: Is this what we want to do here? Does this clean up 100% of the run? If you run selenium tests there may be files left over from the run, we want to stop that

File: src/tests/functional/test_competitions.py (hunk 2 above)
Comment (brendon-lind) [reply src/tests/functional/test_competitions.py:91]: Sweet thanks. I thought it made it easier to parse through.

File: src/static/riot/competitions/editor/_phases.tag (hunk 1 above)
Comment (brendon-lind) [reply src/static/riot/competitions/editor/_phases.tag:68]: Yeah I think that was created for first for another part of the test and then the other selenium attribute was created for another element.

File: src/static/riot/datasets/management.tag [hunk 4]
```diff
@@ -152,19 +152,19 @@
             </div>
 
             <form class="ui form coda-animated {error: errors}" ref="form">
```
Comment (ckcollab) [inline src/static/riot/datasets/management.tag:154]: you could put `selenium='dataset-form'` then do lookups like... `form[selenium="dataset-form"] input[name="name"]` to lookup the name input. This makes it a bit easier to maintain, that way if we add 10 new fields we don't need 10 new `selenium='scoring-new-field-name'`

**Pull request merged**